### ISSUE TITLE
Worse metric depth than V1?

### ISSUE THREAD

**Issue by BaderTim** [DepthAnything/Depth-Anything-V2#26]
Hi there!
First of all, thanks for publishing your updated work. The details captured by this version are amazing.

However, on my own benchmark dataset, which is based on sparse single-distance lidar ground truth, I get far worse results on metric depth (using max depth 64). Like - 3 times as bad. 

Is this expected because of the synthetic data? I thought that using ViT-G as teacher model has solved the distribution shift.

Thanks!

**Comment by LiheYoung** [comment]
Hi, could you please provide more details? E.g., what do you mean by '3 times as bad'?

According to our V2 results on standard benchmarks (Table 2), such as NYUv2 and KITTI, the aligned metric depth should be at least comparable with our V1 models.

**Comment by BaderTim** [comment]
Hi, thanks for your response.

I am using images from my own camera and resize/crop them to the KITTI format (while keeping my fov around 90°). For every image, there are a few single distance points captured, ranging from 6m to 120m in total.

When evaluating on those, V1 produces a RMSE around 7 while V2 achieves only around 19.

I am using the inference codes from run.py in my evaluation code. For V1 I could not define a maximum distance, for V2 I used 128.

**Comment by myasincifci** [comment]
I have a similar experience using the metric_depth fine-tuning code provided in the v2 repo. However, when I use the ZoeDepth code from the v1 repo with v2 weights I get a slightly better result. I am not particularly familiar with the network architecture so I might be mistaken but it seems to me that the metric ft pipeline in v2 only uses a DPT decoder head which is directly trained for metric depth while the ZoeDepth head used in v1 uses metric bins to convert relative to metric. 

@LiheYoung I am still a little confused about the metric fine-tuning because in the v2 paper it is stated that ZoeDepth is used but the code in the repo seems to be missing the metrics bins. Or am I missing something?

**Comment by bluoluo** [comment]
I have discovered a similar issue. The results of fine-tuning with the nyu2 dataset in v1 are similar to those of the open-source model; In v2, I used the vkitti dataset to fine tune the results, which were not very good and differed significantly from the open-source model.
@LiheYoung

**Comment by Double2016** [comment]
I think the author wants to prove the improvement of their dino encoder compared to V1 through a simple DPTHead.However,it's hard for us to reproduce better metric depth finetune result than V1. Has anyone tried replacing DPT head with ZoeDepth head in v2? Can you get better results?

**Comment by myasincifci** [comment]
@Double2016 I'm definitely getting an improvement using zoedepth with my custom dataset over ddp with the v2 weights.

**Comment by freegaussian** [comment]
This question might not related, but I just wanna ask, did the author fix the params of DINOv2 in pertaining?

**Comment by booker-max** [comment]
> I think the author wants to prove the improvement of their dino encoder compared to V1 through a simple DPTHead.However,it's hard for us to reproduce better metric depth finetune result than V1. Has anyone tried replacing DPT head with ZoeDepth head in v2? Can you get better results?

I also want to ask this question, is it possible to follow the method of v1 (depth anything + zoedepth) and load the pre-trained weights of v2

**Comment by booker-max** [comment]
> > I think the author wants to prove the improvement of their dino encoder compared to V1 through a simple DPTHead.However,it's hard for us to reproduce better metric depth finetune result than V1. Has anyone tried replacing DPT head with ZoeDepth head in v2? Can you get better results?
> 
> I also want to ask this question, is it possible to follow the method of v1 (depth anything + zoedepth) and load the pre-trained weights of v2

My answer is yes, you can use the zoe depth training method in v1 and load the weights in v2 at the same time.
